src/UCTNode.cpp

```diff
@@ -258,6 +261,10 @@ bool UCTNode::has_children() const {
 
 void UCTNode::set_visits(int visits) {
     m_visits = visits;
+    if(m_parent != nullptr) {
+        m_parent->add_parent_visit(visits);
```

Comment:
Don't we need to also subtract the old value of `m_visits`?

Reply:
I guess you are right!